openedx/adg/lms/applications/views.py

```diff
@@ -262,4 +264,29 @@ def get_context_data(self, request):
                     'birth_month': extended_profile.birth_date.month,
                     'birth_year': extended_profile.birth_date.year,
                 })
+
+
+class EducationAndExperienceView(RedirectToLoginOrRelevantPageMixin, TemplateView):
+
+    login_url = reverse_lazy('register_user')
+    template_name = 'adg/lms/applications/education_experience.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        user_application = self.request.user.application
+        context.update({
+            'degree_types': Education.DEGREE_TYPES,
+            'month_choices': month_choices(),
+            'year_choices': year_choices(),
+            'user_application_id': user_application.id,
+            'is_work_experience_not_applicable': user_application.is_work_experience_not_applicable
+        })
         return context
+
+    def is_precondition_satisfied(self):
+        try:
+            application_hub = self.request.user.application_hub
+        except ObjectDoesNotExist:
+            return False
+
+        return application_hub.is_written_application_started and not application_hub.is_written_application_completed
```

Comment:
```suggestion
        application_hub = getattr(self.request.user, 'application_hub')
        return application_hub and application_hub.is_written_application_started and not application_hub.is_written_application_completed
```

Reply:
These are'not simple python classes. hasattr deals with AttributeError but not with ObjectDoesNotExist exception. getattr also make use of hasattr.